src/shared/exec-util.c

```diff
@@ -365,8 +366,76 @@ static int gather_environment_consume(int fd, void *arg) {
         return r;
 }
 
+int exec_command_flags_from_strv(char **ex_opts, ExecCommandFlags *flags) {
+        ExecCommandFlags ex_flag = 0;
+        char **opt = NULL;
+
+        assert(flags);
+
+        STRV_FOREACH(opt, ex_opts) {
+                ex_flag = exec_command_flags_from_string(*opt);
+                if (ex_flag != _EXEC_COMMAND_FLAGS_INVALID)
+                        *flags |= ex_flag;
+                else
+                        return -EINVAL;
+        }
+
+        return 0;
+}
+
+int exec_command_flags_to_strv(ExecCommandFlags flags, char ***ex_opts) {
+        ExecCommandFlags it = flags;
+        const char *str;
+        int i, r;
+
+        assert(ex_opts);
+
+        for (i = 0; it != 0; it &= ~(1 << i), i++) {
+                if (FLAGS_SET(flags, (1 << i))) {
+                        str = exec_command_flags_to_string((1 << i));
+                        if (str == NULL)
+                                return -EINVAL;
+
+                        r = strv_extend(ex_opts, str);
```

Comment:
here too: this clobbers the return value on failure. Please instead build up the full list locally in some local variable, then return it when it is done at the end.